engine/proofview.ml

```diff
@@ -700,9 +715,9 @@ let mark_in_evm ~goal evd content =
             | loc,_ -> loc,Evar_kinds.GoalEvar }
     else info
   in
-  let info = match Evd.Store.get info.Evd.evar_extra typeclass_resolvable with
-  | None -> { info with Evd.evar_extra = Evd.Store.set info.Evd.evar_extra typeclass_resolvable () }
-  | Some () -> info
+  let info = match get_typeclass_unresolvable info.Evd.evar_extra with
+  | None -> { info with Evd.evar_extra = set_typeclass_unresolvable info.Evd.evar_extra }
+  | Some _ -> info
```

Comment:
Why did that change from `()`? (it doesn't change the generated code, it is just an guarantee that we're actually matching something of type `unit`)